### PR TITLE
Adding a CA usage example to README.md

### PR DESCRIPTION
**Pull request by sourcequench**
How to actually create grants/identities and sign certificates with these probably not is immediately obvious, so an example is probably useful for those wanting to go beyond the local setup shell script.

### CODE REVIEW

File: README.md
Comment (blunderer) [inline README.md:149]: You can't just copy keys, but you can import a user existing pubkey (no need for the private key) by running:

hiba-ca.sh -i -u -f path_to_pubkey -I principal

File: README.md
Comment (blunderer) [inline README.md:158]: The keys you generated right above are user keys, not host keys. OpenSSH makes the difference between the two types.

If you meant to retrieve the key from the host, you should add a step, otherwise it might confuse people. Also, the import has to be done with higba-ca.sh -i -h -f path_to_pubkey -I hostname

File: README.md
Comment (blunderer) [inline README.md:180]: is the hostname that that you set when importing the host pubkey.

File: README.md
Comment (blunderer) [inline README.md]: The hostname is automatically passed as principal, -n is for adding more principals, and it is not required here.

File: README.md
Comment (blunderer) [inline README.md]: Not needed in interactive mode, this is to pass the encryption key to ssh-keygen to unlock the CA signing key, but here the user can just type it.

File: README.md
Comment (blunderer) [inline README.md:198]: Maybe add the commands for doing this?

ssh-add /tmp/hiba-ca/users/testuser-cert.pub
ssh user@hostname

File: README.md
Comment (blunderer) [inline README.md:196]: It also needs the TrustedCAKeys option to be pointing at the CA pubkey (that has to be copied to the host once).

Maybe point at the configuration documentation: https://github.com/google/hiba/blob/main/INSTALL.md#configuration

File: README.md
Comment (blunderer) [inline README.md:132]: It might actually make more sense to create a separate file here, and reference it in the README (sorry for coming back on my first idea).

The rationale:
- the example usage is longer than the rest of the file itself
- hiba-ca.sh is just one way of dealing with HIBA, there can be other CA integrations.

**Review by blunderer — COMMENTED** [review]
Please merge your two commits into one, until this makes it to the main branch, it is fine to rewrite history to make it look clean and cleare.

File: README.md
Comment (sourcequench) [reply README.md:141]: I'll work on the history revision - my git knowledge is limited so I'll have to go look that up. Changed everything to remove -d and instead use ~/.hiba-ca.

File: README.md
Comment (sourcequench) [reply README.md:149]: updated - first in this file but I'm going to move everything to its own md file and out of readme.

File: README.md
Comment (sourcequench) [reply README.md:158]: I changed to "Copy the OpenSSH server host public key from the OpenSSH server"...

My immediate hope is for this to be useful to vendors of networking devices, so how that is done differs based on the platform.

It all works fine for me simply copying files to the hosts structure with devicename.pub - I can sign those into certificates with the right identity information. That seems less steps than copying down the file to some path, keeping track of that, then using hiba-ca.sh to import. What am I missing?

File: README.md
Comment (sourcequench) [reply README.md:198]: My expectation is that an explicit -I is a more reliable plan than expecting people have agents - and at least for my immediate target audience there are even odds they are using putty or similar as a client :(

File: README.md
Comment (blunderer) [reply README.md]: You can run `git rebase -i HEAD~N` with N the number of revisions you want to revisit.

File: README.md
Comment (blunderer) [reply README.md]: LGTM

File: README.md
Comment (sourcequench) [reply README.md]: I did that, and then learned I need to prefix each of the commits with a "squash" word in the editor - so `git log` looks good locally, but I'm not able to push to origin because "Updates were rejected because the tip of your current branch is behind its remote counterpart". The fast forward makes me nervous, and now if I try to git pull I get a hint about divergent branches and setting git config stuff - so I set the strategy to merge and re-pulled down, which just gave me back my prior commits and added a new one for the merge - so I assume I am missing some --just-replace-origin-plz command.

Comment (blunderer) [reply README.md]: The answer here is git push -f origin <branch> 

The -f is a problem on the main branch (or any public branch), but it is perfectly fine in your dev branches.
If you worry about an accidental bad push, don't: the main branch is protected against direct pushes for everyone (including maintainers), and only can be updated via the review process.

Comment (sourcequench) [reply README.md]: done.

File: CA.md
Comment (blunderer) [inline CA.md:16]: This is not needed, ssh-keygen will use an empty passphrase if the user presses [enter] at prompt time.

Since I suspect some people will be using these instructions not just for test, it would be better to give them the choice of using a secret passphrase or not, rather than making the call they do not need one.

Comment (blunderer) [reply CA.md:16]: then you can also leave the `-- -N "" ` from all other commands, making them simpler. This -N is mostly needed when using in non-interactive more like in the tests.

File: CA.md
Comment (blunderer) [inline CA.md:20]: Please remove, this is now already addressed below.

File: CA.md
Comment (blunderer) [inline CA.md:30]: Maybe add a new code block with the actual command and corresponding stdout for consistency.

File: CA.md
Comment (blunderer) [inline CA.md:20]: s/test//

File: CA.md
Comment (blunderer) [inline CA.md:32]: Can you also add the command for generating the keys similarly to the step above?

File: CA.md
Comment (blunderer) [inline CA.md:38]: the '~' will not be expanded to home directory when enclosed in double quotes, you should remove them here.

Comment (blunderer) [reply CA.md:38]: And in other commands below

File: CA.md
Comment (blunderer) [inline CA.md:50]: "points to"

File: CA.md
Comment (blunderer) [inline CA.md:54]: Not needed, the hostname is added as a principal already, this serves for adding extra principals (maybe the host IPv4 / IPv6 addresses).

Comment (blunderer) [inline CA.md:54]: You might want to set the validity which defaults to 1h. This is fine for users, but will be toily for hosts:

$ ./hiba-ca.sh -s -h -I test -H prod -v +12w

For 12 weeks (~3 months) validity.

File: CA.md
Comment (blunderer) [inline CA.md:66]: This rmd.me/iigY1QeGBKr seems suspicious

File: CA.md
Comment (blunderer) [inline CA.md:73]: This should be a new step and you can add the actual `ssh -I` command for new users not knowing how to select an identity for sshing into a target.

File: CA.md
Comment (sourcequench) [reply CA.md:37]: Added in an example of generating a host keypair. I expect the most common pattern will be to import a public key from the device, so I've put the fresh keypair generation under "Alternatively"

File: CA.md
Comment (blunderer) [inline CA.md:68]: My bad, the `-v` is for verbose. Validity is `-V` (capital v)

Comment (sourcequench) [reply CA.md]: done.

File: README.md
Comment (blunderer) [reply README.md]: The eligibility requires a specific folder named after the principal, which is created when either creating keys or importing them using the hiba-ca.sh command.

I would also avoid external assumptions on hard-coded paths except for the HIBA policy itself.

File: CA.md
Comment (blunderer) [inline CA.md:76]: This should not be used here (same rationale as for the CA keys)

File: CA.md
Comment (blunderer) [inline CA.md:55]: nit: maybe put in backquotes?

Create a grant called `shell` with a matching domain

File: CA.md
Comment (sourcequench) [reply CA.md:65]: sorry, not following the change you want on this one.

Comment (blunderer) [reply CA.md:65]: Sorry, I meant to maybe replace:

-I is the hostname you

by 

-I points to the hostname you

But on second though, I am not so sure anymore.  I'll leave it up to you to decide which one works best

File: README.md
Comment (sourcequench) [reply README.md]: Interesting though on the folder name - no idea how I got it to work in my testing without the import.

Comment (blunderer) [reply README.md]: There is no "eligible"  for hosts, so it will most likely work, but not for users.

**Review by blunderer — APPROVED** [review]

**Pull request merged**
